content/client-lib-development-guide/features.textile

```diff
@@ -606,6 +612,13 @@ h4. ClientOptions
 *** @(TO3k3)@ @realtimeHost@ string - for development environments only; allows a non-default Ably Realtime host to be specified
 *** @(TO3k4)@ @port@ integer - for development environments only; allows a non-default Ably non-TLS port to be specified
 *** @(TO3k5)@ @tlsPort@ integer - for development environments only; allows a non-default Ably TLS port to be specified
+** @(TO3l)@ The follow attributes, if set, are used to change the default behaviour of the libary:
+*** @(TO3l1)@ @disconnectedRetryFrequency@ integer - default 15,000 (15s). When the connection is in the @DISCONNECTED@ state, this delay in milliseconds controls how frequently the client library attempts to reconnect automatically
+*** @(TO3l2)@ @suspendedRetryFrequency@ integer - default 30,000 (30s). When the connection is in the @SUSPENDED@ state, this delay in milliseconds controls how frequently the client library attempts to reconnect automatically
+*** @(TO3l3)@ @httpOpenTimeout@ integer - default 4,000 (4s). Timeout for opening the connection, available in the client library if supported by the transport
```

Comment:
> most HTTP client libraries that I took a quick look at supported separate timeout values for connect & response.

That's reasonable for a generic library because you might have a request that you expect to be long-running (eg a comet recv) but you'd still want to know very quickly if it had failed to send the request. In the defaults we had in the ably-js library there was a single composite timeout (the connection + the response) for "normal" requests, and a different timeout for "long" requests. For the latter, I'd expect the library could still raise an error if the connection and the sending of the request hadn't occurred within the "normal" time.